enterprise_access/apps/subsidy_access_policy/models.py

```diff
@@ -591,10 +592,29 @@ def can_redeem(self, lms_user_id, content_key, skip_customer_user_check=False):
 
         # learner not associated to enterprise
         if not skip_customer_user_check:
-            if self.lms_api_client.get_enterprise_user(self.enterprise_customer_uuid, lms_user_id) is None:
+            learner_record = self.lms_api_client.get_enterprise_user(self.enterprise_customer_uuid, lms_user_id)
+            if not learner_record:
                 self._log_redeemability(False, REASON_LEARNER_NOT_IN_ENTERPRISE, lms_user_id, content_key)
                 return (False, REASON_LEARNER_NOT_IN_ENTERPRISE, [])
 
+            associated_group_uuids = set(  # pylint: disable=consider-using-set-comprehension
+                [str(group.get('uuid')) for group in learner_record.get('enterprise_group')]
```

Comment:
I would add a default value for the learner_record getter- otherwise a None value returned may be a problem:
```
>>> yo = [foo for foo in ['b', 'a', 'r']]
>>> yo
['b', 'a', 'r']
>>> yo_2 = [foo for foo in None]
Traceback (most recent call last):
  File "<console>", line 1, in <module>
TypeError: 'NoneType' object is not iterable
```